BrainUp/BrainUp/basics/CommonViews/GradientButton.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  GradientButton.swift
+//  BrainUp
+//
+//  Created by Andrei Trukhan on 17/02/2022.
+//
+
+import UIKit
+
+class GradientButton: UIButton {
+
+    private var cornerRadius: CGFloat = 16
+    private var insets: UIEdgeInsets = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setupUI()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.setupUI()
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.setupUI()
+    }
+
+    func update(cornerRadius: CGFloat, insets: UIEdgeInsets) {
+        self.cornerRadius = cornerRadius
+        self.insets = insets
+    }
+
+    private func setupUI() {
+        setTitleColor(.white, for: [.normal, .highlighted, .selected, .disabled])
+        titleLabel?.font = UIFont.montserratSemiBold(size: 12)
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        layer.anchorPoint = CGPoint(x: 0.5, y: 0.5)
+
+        layer.borderWidth = 1.0
+        layer.cornerRadius = cornerRadius
+        layer.borderColor = UIColor.coldViolet.cgColor
+        contentEdgeInsets = insets
+
+        backgroundColor = UIColor.clear
+
+        let gradient = CAGradientLayer()
```

Comment:
If I use your function, the border of button will be bigger than gradient. It is happen because your gradient method uses start point 0.25, end point 0.75 and some kind of transformation.